src/domain.rs

```diff
@@ -47,6 +47,18 @@ impl Global {
     }
 }
 
+/// Mark a [domain family](Domain) that unique caracterize a [domain instance](Domain)
+///
+/// See [`Global`] and [`unique_domain`]
+///
+/// # Safety
+///
+/// Implementors of this trait must guarantee only one Domain of that family can be contructed.
+pub unsafe trait Singleton {}
+
+// Safety: there are only one instance of Domain<Global>
```

Comment:
I'd like to see a bit more documentation for this safety check. Specifically, we can guarantee that there's only ever one `Domain<Global>` because `Global` itself is not possible to construct outside of this crate (due to `#[non_exhaustive]` + no public constructor), and we only ever construct one `Domain` from it internally in the form of a single `static`.

Reply:
I will improve the docs, tanks to the tip.